ffi/src/lib.rs

```diff
@@ -50,7 +50,10 @@ use tokio::runtime::Runtime;
 /// * `script_bytes` must point to `script_len` consecutive properly initialized bytes.
 /// * The returned C string must be freed by the caller.
 #[no_mangle]
-pub unsafe extern "C" fn store_script(script_bytes: *const u8, script_len: usize) -> *mut c_char {
+pub unsafe extern "C-unwind" fn store_script(
+    script_bytes: *const u8,
+    script_len: usize,
+) -> *mut c_char {
     let script = unsafe { std::slice::from_raw_parts(script_bytes, script_len) };
     let hash = scripts_container::add_script(script);
     CString::new(hash).unwrap().into_raw()
```

Comment:
What we're doing here is UB and needs to be fixed. Rust allocated memory needs to be cleaned up by Rust code. We also potentially panic here because we try to convert to a C string and back to a Rust string without knowing if there are Nul bytes in the string. @asafpamzn suggested we return binary data here instead. This seems like a reasonable change, but as long as glide-core stores hashes as Rust `String`s, we can't make that change without changing every other language's wrapper code. I'll be making those changes in a part 2 to this PR.

Reply:
After reading this code again, I'm pretty sure we don't need to change all wrappers. We can store hashes as Rust strings just fine, it's just that they need to be converted to binary data instead of C strings when passed across the FFI boundary and back. I'll try making this change in a separate branch for now and if the changes aren't too large, I'll add it to this PR.